pkg/operator/mds/controller.go

```diff
@@ -101,3 +106,19 @@ func (c *FilesystemController) onDelete(obj interface{}) {
 		logger.Errorf("failed to delete file system %s. %+v", filesystem.Name, err)
 	}
 }
+
+func filesystemChanged(oldFS, newFS FilesystemSpec) bool {
+	if len(oldFS.DataPools) != len(newFS.DataPools) {
```

Comment:
is there a param or configuration that we shouldnt allow to change? If so we should have it here and perhaps issue an error

Reply:
All we could do is log an error. We can't abort the crd update. So my approach was to only detect the conditions we support upgrading and ignore all the other combinations.